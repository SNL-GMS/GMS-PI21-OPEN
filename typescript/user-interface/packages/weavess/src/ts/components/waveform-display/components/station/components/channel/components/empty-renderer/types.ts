import type { WeavessTypes } from '@gms/weavess-core';

export interface EmptyRendererProps {
  /** waveform interval loaded and available to display */
  displayInterval: WeavessTypes.TimeRange;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyRendererState {}
