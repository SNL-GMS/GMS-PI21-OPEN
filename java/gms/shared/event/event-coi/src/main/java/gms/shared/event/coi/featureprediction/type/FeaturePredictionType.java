package gms.shared.event.coi.featureprediction.type;

import com.fasterxml.jackson.annotation.JsonValue;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
import gms.shared.event.coi.featureprediction.value.ArrivalTimeFeaturePredictionValue;
import gms.shared.event.coi.featureprediction.value.FeaturePredictionValue;
import gms.shared.event.coi.featureprediction.value.NumericFeaturePredictionValue;

import java.io.IOException;
import java.util.Map;
import java.util.Optional;

/**
 * Class for an instantiable object that represents the type of feature prediction. Each "type"
 * is associated with a FeaturePredictionValue class; together, they fully define what each type of
 * feature prediction looks like.
 *
 * @param <T> The FeaturePredictionValue class that should be coupled with this type.
 */
@JsonDeserialize(using = FeaturePredictionType.Deserializer.class)
public abstract class FeaturePredictionType<T extends FeaturePredictionValue<?, ?, ?>> {

  public enum TypeNames {
    ARRIVAL_TIME,
    SLOWNESS,
    EMERGENCE_ANGLE,
    RECEIVER_TO_SOURCE_AZIMUTH,
    SOURCE_TO_RECEIVER_AZIMUTH,
    SOURCE_TO_RECEIVER_DISTANCE,
  }

  private final TypeNames name;

  private FeaturePredictionType(
    TypeNames name
  ) {
    this.name = name;
  }

  @JsonValue
  public TypeNames getName() {
    return name;
  }

  @Override
  public String toString() {
    return "FeaturePredictionType{" +
      "name='" + name + '\'' +
      '}';
  }

  public abstract Class<T> getTypeValueClass();

  //
  // Constants for the supported feature prediction types.
  //

  public static final FeaturePredictionType<ArrivalTimeFeaturePredictionValue> ARRIVAL_TIME_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.ARRIVAL_TIME) {
      @Override
      public Class<ArrivalTimeFeaturePredictionValue> getTypeValueClass() {
        return ArrivalTimeFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> SLOWNESS_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.SLOWNESS) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> EMERGENCE_ANGLE_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.EMERGENCE_ANGLE) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> RECEIVER_TO_SOURCE_AZIMUTH_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.RECEIVER_TO_SOURCE_AZIMUTH) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> SOURCE_TO_RECEIVER_AZIMUTH_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.SOURCE_TO_RECEIVER_AZIMUTH) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> SOURCE_TO_RECEIVER_DISTANCE_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.SOURCE_TO_RECEIVER_DISTANCE) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  /**
   * Custom deserializer used by Jackson. The serialized JSON should contain a single field with
   * a single string that is the name of one of the types, and that string will be returned
   * be JsonParser::getValueAsString.
   */
  static class Deserializer extends StdDeserializer<FeaturePredictionType<?>> {

    private static final Map<TypeNames, FeaturePredictionType<?>> nameTypeMap = Map.of(
      TypeNames.ARRIVAL_TIME, ARRIVAL_TIME_PREDICTION_TYPE,
      TypeNames.SLOWNESS, SLOWNESS_PREDICTION_TYPE,
      TypeNames.EMERGENCE_ANGLE, EMERGENCE_ANGLE_PREDICTION_TYPE,
      TypeNames.RECEIVER_TO_SOURCE_AZIMUTH, RECEIVER_TO_SOURCE_AZIMUTH_PREDICTION_TYPE,
      TypeNames.SOURCE_TO_RECEIVER_AZIMUTH, SOURCE_TO_RECEIVER_AZIMUTH_PREDICTION_TYPE,
      TypeNames.SOURCE_TO_RECEIVER_DISTANCE, SOURCE_TO_RECEIVER_DISTANCE_PREDICTION_TYPE
    );

    public Deserializer() {
      this(null);
    }

    public Deserializer(Class<?> vc) {
      super(vc);
    }

    @Override
    public FeaturePredictionType<?> deserialize(JsonParser p, DeserializationContext ctxt)
      throws IOException {
      return Optional.ofNullable(nameTypeMap.get(TypeNames.valueOf(p.getValueAsString())))
        .orElseThrow();
    }
  }
}
