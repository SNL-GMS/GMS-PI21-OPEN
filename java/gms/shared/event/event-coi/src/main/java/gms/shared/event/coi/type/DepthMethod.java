package gms.shared.event.coi.type;

import com.google.common.base.Functions;
import com.google.common.base.Preconditions;
import com.google.common.collect.ImmutableMap;

import java.util.Arrays;

public enum DepthMethod {
  A("a"),
  F("f"),
  D("d"),
  R("r"),
  G("g"),
  UNKNOWN("-");


  private static final ImmutableMap<String, DepthMethod> LABELS_BY_MODEL;

  static {
    LABELS_BY_MODEL = Arrays.stream(values())
      .collect(ImmutableMap.toImmutableMap(DepthMethod::getLabel, Functions.identity()));
  }

  public final String label;


  DepthMethod(String label) {
    this.label = label;
  }

  public static DepthMethod valueOfLabel(String label) {
    Preconditions.checkNotNull(label, "label cannot be null");
    Preconditions.checkArgument(!label.isBlank(), "label cannot be blank");
    return LABELS_BY_MODEL.get(label);
  }

  public String getLabel() {
    return label;
  }

}

