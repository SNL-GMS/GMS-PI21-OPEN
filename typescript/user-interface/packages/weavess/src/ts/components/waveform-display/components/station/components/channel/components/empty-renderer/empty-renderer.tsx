/* eslint-disable react/destructuring-assignment */
import React from 'react';
import * as THREE from 'three';

import type { EmptyRendererProps, EmptyRendererState } from './types';

/**
 * Empty component. Renders and displays an empty graphics data.
 */
export class EmptyRenderer extends React.PureComponent<EmptyRendererProps, EmptyRendererState> {
  /** THREE.Scene for this channel */
  public scene: THREE.Scene;

  /** Orthographic camera used to zoom/pan around the spectrogram */
  public camera: THREE.OrthographicCamera;

  /** Current min in gl units */
  private readonly glMin = 0;

  /** Current max in gl units */
  private readonly glMax = 100;

  /**
   * Constructor
   *
   * @param props props as SpectrogramRendererProps
   */
  public constructor(props: EmptyRendererProps) {
    super(props);
    this.state = {};
  }

  // ******************************************
  // BEGIN REACT COMPONENT LIFECYCLE METHODS
  // ******************************************

  /**
   * Called immediately after a component is mounted.
   * Setting state here will trigger re-rendering.
   */
  public componentDidMount(): void {
    this.scene = new THREE.Scene();
    const cameraZDepth = 5;
    this.camera = new THREE.OrthographicCamera(
      this.glMin,
      this.glMax,
      1,
      -1,
      cameraZDepth,
      -cameraZDepth
    );
    this.camera.position.z = 0;
  }

  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, react/sort-comp
  public render() {
    return null;
  }
}
