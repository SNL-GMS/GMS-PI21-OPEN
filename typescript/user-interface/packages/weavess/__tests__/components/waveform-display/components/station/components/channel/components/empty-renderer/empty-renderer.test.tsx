/* eslint-disable react/jsx-props-no-spreading */
import { render } from '@testing-library/react';
import React from 'react';

import { EmptyRenderer } from '../../../../../../../../../src/ts/components/waveform-display/components/station/components/channel/components/empty-renderer/empty-renderer';
import type { EmptyRendererProps } from '../../../../../../../../../src/ts/components/waveform-display/components/station/components/channel/components/empty-renderer/types';

const props: EmptyRendererProps = {
  displayInterval: {
    startTimeSecs: 0,
    endTimeSecs: 500
  }
};

describe('Weavess Empty Renderer', () => {
  it('to be defined', () => {
    expect(EmptyRenderer).toBeDefined();
  });

  it('renders', () => {
    const { container } = render(<EmptyRenderer {...props} />);
    expect(container).toMatchSnapshot();
  });
});
