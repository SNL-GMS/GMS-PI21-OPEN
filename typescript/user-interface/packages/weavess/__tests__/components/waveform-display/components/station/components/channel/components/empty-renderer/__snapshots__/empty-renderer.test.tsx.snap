// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess Empty Renderer renders 1`] = `<div />`;
